NatTypes.dfy(10:4-10:4): Error: value does not satisfy the subset constraints of 'nat'
NatTypes.dfy(35:11-35:11): Error: value does not satisfy the subset constraints of 'nat'
NatTypes.dfy(45:6-45:6): Error: assertion could not be proved
NatTypes.dfy(48:6-48:6): Error: assertion could not be proved
NatTypes.dfy(51:6-51:6): Error: assertion could not be proved
NatTypes.dfy(54:6-54:6): Error: assertion could not be proved
NatTypes.dfy(65:6-65:6): Error: assertion could not be proved
NatTypes.dfy(71:6-71:6): Error: assertion could not be proved
NatTypes.dfy(74:6-74:6): Error: assertion could not be proved
NatTypes.dfy(91:6-91:6): Error: assertion could not be proved
NatTypes.dfy(105:6-105:6): Error: assertion could not be proved
NatTypes.dfy(126:21-126:21): Error: value does not satisfy the subset constraints of 'nat'
NatTypes.dfy(141:44-141:44): Error: value does not satisfy the subset constraints of 'nat'
NatTypes.dfy(164:34-164:34): Error: value does not satisfy the subset constraints of 'nat'
NatTypes.dfy(182:18-182:18): Error: result of operation could violate subset type constraint for 'nat'
NatTypes.dfy(190:19-190:19): Error: value of expression (of type 'GList<nat>') is not known to be an instance of type 'GList<int>'

Dafny program verifier finished with 6 verified, 16 errors
